Commit ID: 628f50601b9b4b1988f52f6d599b7d94efb81a89
Change ID: ovmwlzvksrwsmtkqxrrnlxlyrzqkyvzx
Author   : flukey <flukey@vapourmail.eu> (2026-10-05 15:34:16)
Committer: flukey <flukey@vapourmail.eu> (2026-10-05 16:04:37)

    udev no systemd

diff --git a/udev/69-dm-lvm.rules.in b/udev/69-dm-lvm.rules.in
index 432049a5df..25ab610ff2 100644
--- a/udev/69-dm-lvm.rules.in
+++ b/udev/69-dm-lvm.rules.in
@@ -89,8 +89,14 @@
 # it's better suited to appearing in the journal.
 
 IMPORT{program}="(LVM_EXEC)/lvm pvscan --cache --listvg --checkcomplete --vgonline --autoactivation event --udevoutput --journal=output $env{DEVNAME}"
+TEST!="/run/systemd/system", GOTO="lvm_direct_vgchange"
+
 ENV{LVM_VG_NAME_COMPLETE}=="?*", RUN+="(SYSTEMDRUN) --no-block --property DefaultDependencies=no --unit lvm-activate-$env{LVM_VG_NAME_COMPLETE} (LVM_EXEC)/lvm vgchange -aay --autoactivation event $env{LVM_VG_NAME_COMPLETE}"
 GOTO="lvm_end"
 
+LABEL="lvm_direct_vgchange"
+ENV{LVM_VG_NAME_COMPLETE}=="?*", RUN+="(LVM_EXEC)/lvm vgchange -aay --autoactivation event $env{LVM_VG_NAME_COMPLETE}"
+GOTO="lvm_end"
+
 LABEL="lvm_end"
 
